### PR TITLE
Integrate GF 180 MCU PDK

### PR DESCRIPTION
**Pull request by ThomasPluck**
Introduces new base primitive `FourTerminalBipolar` and modifies `sitepdks` as needed. The format of spice files in the GF180 tech files are different and the `Install` class in `pdk.py` reflects that.

Otherwise, all new components tested for netlisting, compilation and simulation.

### CODE REVIEW

File: hdl21/primitives.py [hunk 1]
```diff
@@ -400,6 +405,25 @@ class PhysicalCapacitorParams:
     aliases=["Cap3", "PhyCap3", "CapPhy3", "PhyCapacitor3"],
 )
 
+# For the 4 terminal components, we have
```
Comment (dan-fritchman) [inline hdl21/primitives.py:408]: Same comment as for Sky130: I don't think we should have this as a generic. 

There's too much interpretation to be had in what the "other two" terminals mean.

File: pdks/Gf180/gf180/pdk.py [hunk 2]
```diff
@@ -0,0 +1,560 @@
...
+            return self.bjt_module_call(call.params)
+
+        else:
+            raise RuntimeError(f"{call.prim} is not legitimate primitive")
+
+        return call
+
+    def mos_module(self, params: MosParams) -> h.ExternalModule:
+        """Retrieve or create an `ExternalModule` for a MOS of parameters `params`."""
+        mod = xtors.get((params.model, params.tp), None)
```
Comment (dan-fritchman) [inline pdks/Gf180/gf180/pdk.py:403]: Similar comment to the Sky130 PR here - it seems the implication is that every generic primitive _must_ set a `model` field to be compiled into this tech. 

While I haven't used this technology, there should, in general, be some notion of "the regular transistors" - i.e. the ones the standard logic library uses. The ideal of having different thresholds `vth` also generally transfers between techs. We should reserve the `model` field as an escape hatch for "weirder", more tech-specific things - e.g. the difference between having 6V and 20V devices, when the "main" ones are 1.8V.

**Comment by ThomasPluck** [comment]
Brought up to similar standards as SKY130 at the moment, should resolve all criticisms.

**Comment by ThomasPluck** [comment]
Going to repeat this in a PR which includes the changes and lessons learnt from integrating SKY130.

**Pull request closed without merging**